pager/pager.c

```diff
@@ -2513,6 +2522,17 @@ int mutt_pager(struct PagerView *pview)
   //-------------------------------------------------------------------------
   while (op != -1)
   {
+    if (delay_begin != 0)
+    {
+      const uint64_t now = mutt_date_epoch_ms();
+      if (now - delay_begin > c_pager_read_delay * 1000)
+      {
+        mutt_debug(LL_DEBUG4, "delay elapsed while still on message, marking message read\n");
+        mutt_set_flag(m, pview->pdata->email, MUTT_READ, true);
+        delay_begin = 0;
+      }
+    }
+
```

Comment:
Do you actually need this block both when entering and when exiting the `while (op != -1)` loop?

Reply:
`mutt_pager` is a huge function, would you mind factoring your block into a dedicated function?

Reply:
> `mutt_pager` is a huge function, would you mind factoring your block into a dedicated function?

Sure, will do in v2.

Reply:
> Do you actually need this block both when entering and when exiting the `while (op != -1)` loop?

I think so. Not all actions navigate away from a message, so it is nicer to make the switch at the first possible action after enough time has elapsed if that will be observable by later actions while still in the pager.  But if you think otherwise, that a single check after the while() loop exits is sufficient, that's an easy enough change.  I'm also thinking towards the future: right now, the 'N' indicator in the index line (if you have pager_index_lines set) remains displayed.  Cooler would be having the 'N' indicator automatically clear itself at the first redraw after the message has been marked new, which requires clearing the unread bit inside the while loop.